src/Microsoft.Graph.Core/Requests/RedirectHandler.cs

```diff
@@ -0,0 +1,172 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net.Http;
+    using System.Text;
+    using System.Net.Http.Headers;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using System.Net;
+    
+
+    /// <summary>
+    /// An <see cref="DelegatingHandler"/> implementation using standard .NET libraries.
+    /// </summary>
+    public class RedirectHandler : DelegatingHandler
+    {
+
+        private const int maxRedirects = 5;
+
+        /// <summary>
+        /// Constructs a new <see cref="RedirectHandler"/> 
+        /// </summary>
+        /// <param name="innerHandler">An HTTP message handler to pass to the <see cref="HttpMessageHandler"/> for sending requests.</param>
+        public RedirectHandler(HttpMessageHandler innerHandler)
+        {
+            InnerHandler = innerHandler;
+        }
+
+        /// <summary>
+        /// Disposes the HttpMessageHandler instance.
+        /// </summary>
+        public new void Dispose()
+        {
+            if (InnerHandler != null)
```

Comment:
Don't dispose what you don't own/create.  InnerHandler was passed into the constructor, it's lifetime should not be tied to the lifetime of this class.